flexmeasures/data/models/planning/battery.py

```diff
@@ -71,14 +71,18 @@ def schedule_battery(
         )  # shift "equals" constraint for target SOC by one resolution (the target defines a state at a certain time,
         # while the "equals" constraint defines what the total stock should be at the end of a time slot,
         # where the time slot is indexed by its starting time)
-    device_constraints[0]["min"] = (asset.min_soc_in_mwh - soc_at_start) * (
-        timedelta(hours=1) / resolution
+    device_constraints[0]["min"] = (
+        sensor.generic_asset.get_attribute("min_soc_in_mwh") - soc_at_start
```

Comment:
So this will raise if the `Sensor` does not have the necessary attributes.

Is there a good way to check? For instance, we could check the GenericAssetType. Or make one quick check if all attributes we require are present...

Reply:
The planner can work without some of the attributes, such as `is_pure_consumer`; others it can set a default for, such as `min_soc_in_mwh` with default 0; yet others it cannot realistically work without, such as `capacity_in_mw`.

My next PR https://github.com/SeitaBV/flexmeasures/issues/242 introduces the `has_attribute` method on the `Sensor` and `GenericAsset` classes, which will be useful for this purpose. I'll make a project ticket to go over the planners and set defaults or raise where needed (see https://github.com/SeitaBV/flexmeasures/issues/245).